### PR TITLE
[WIP] MON-2901: add nodeExporter.collectors.netdev settings.

### PR DESCRIPTION
**Pull request by raptorsun**
* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Review by simonpasquier — COMMENTED** [review]
Unless required, I'd move 05358f2127ddf3375b4755204673a076e6a719a7 to a separate PR to facilitate the review.

File: pkg/manifests/types.go [hunk 1]
```diff
@@ -286,6 +289,14 @@ type NodeExporterCollectorTcpStatConfig struct {
 	Enabled bool `json:"enabled,omitempty"`
 }
 
+// The `NodeExporterCollectorNetDevConfig` resource works as an on/off switch for
+// the `netdev` collector of the `node-exporter` agent.
+// By default, the `netdev` collector is enabled.
+type NodeExporterCollectorNetDevConfig struct {
+	// A Boolean flag that enables or disables the `netdev` colletor.
```
Comment (simonpasquier) [inline pkg/manifests/types.go:296]: We need to document clearly which metrics will be missing hwen the collector is disabled.

**Comment by raptorsun** [comment]
> Unless required, I'd move [05358f2](https://github.com/openshift/cluster-monitoring-operator/commit/05358f2127ddf3375b4755204673a076e6a719a7) to a separate PR to facilitate the review.

Good point, I will create a  separate PR for the improvement of default value settings and rebase this PR on it 🫡

**Comment by simonpasquier** [comment]
@raptorsun I'd do it the other way around: the refactoring of the default values can come after this PR :)

**Comment by raptorsun** [comment]
> @raptorsun I'd do it the other way around: the refactoring of the default values can come after this PR :)

ok, I will leave only the bare minimal refactoring in the default value setting, to make `nodeExporter.collectors.netdev.enabled` have true as its default value, in this PR.
The default value setting for other fields in the config structure in a follow-up PR.

**Comment by raptorsun** [comment]
/close
this PR in favor of 
https://github.com/openshift/cluster-monitoring-operator/pull/1888

**Pull request closed without merging**